tests/e2e-evm/test/abi_basic.test.ts

```diff
@@ -0,0 +1,162 @@
+            const txData = { to: mockAddress, data: funcSelector, gas: 25000n, value: 1n };
+
+            const txHash = await walletClient.sendTransaction(txData);
+            const txReceipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+            expect(txReceipt.status).to.equal(isPayable ? "success" : "reverted");
+          });
+
+          it("can be called with extra data", async function () {
+            const data = concat([funcSelector, "0x01"]);
+            const txData = { to: mockAddress, data: data, gas: 25000n };
+
+            const txHash = await walletClient.sendTransaction(txData);
+            const txReceipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+            expect(txReceipt.status).to.equal("success");
+          });
+        });
+      }
+
+      //
+      // Test ABI special functions -- receive & fallback
+      //
+      // Receive functions are always payable and can not receive data
+      // Fallback functions can be payable or non-payable and can receive data in both cases
+      const testName = `ABI special functions: ${receiveFunction ? "" : "no "}receive and ${fallbackFunction ? fallbackFunction.stateMutability : "no"} fallback`;
+      describe(testName, function () {
+        if (!receiveFunction && (!fallbackFunction || fallbackFunction.stateMutability !== "payable")) {
+          it("can not receive plain transfers", async function () {
+            const txData = { to: mockAddress, gas: 25000n, value: 1n };
+
+            const txHash = await walletClient.sendTransaction(txData);
+            const txReceipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+            expect(txReceipt.status).to.equal("reverted");
+          });
+        }
+
+        if (receiveFunction || (fallbackFunction && fallbackFunction.stateMutability === "payable")) {
+          it("can receive plain transfers", async function () {
+            const txData = { to: mockAddress, gas: 25000n, value: 1n };
+
+            const txHash = await walletClient.sendTransaction(txData);
+            const txReceipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+            expect(txReceipt.status).to.equal("success");
+          });
+        }
+
+        it(`can ${fallbackFunction ? "" : "not "}be called with an invalid function selector`, async function () {
+          const data = toFunctionSelector("does_not_exist()");
+          const txData = { to: mockAddress, data: data, gas: 25000n };
+
+          const txHash = await walletClient.sendTransaction(txData);
+          const txReceipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+          expect(txReceipt.status).to.equal(fallbackFunction ? "success" : "reverted");
+        });
+
+        if (fallbackFunction) {
+          it(`can ${fallbackFunction.stateMutability === "payable" ? "" : "not "}receive transfer with data or invalid function selector`, async function () {
```

Comment:
I should create a helper here -- `isPayable(fallbackFunction)` that can be used with an ABI defined function